src/pybind/mgr/mgr_module.py

```diff
@@ -311,18 +318,24 @@ def call(self, mgr, cmd_dict, inbuf):
         for a, d in self.args_dict.items():
             if 'req' in d and d['req'] == "false" and a not in cmd_dict:
                 continue
-
-            if isinstance(cmd_dict[a], str) and '=' in cmd_dict[a]:
-                k, arg = cmd_dict[a].split('=', 1)
-                if k in self.args_dict:
+            if 'kw' in d and d['kw'] == 'true':
+                # parse the next arg as a kwarg, but only if it contains an '='
+                if not kwargs_switch and isinstance(cmd_dict[a], str) and '=' in cmd_dict[a]:
+                    # We consider this a kwarg. Set the switch so that all
+                    # future args are parsed as kwargs too.
+                    mgr.log.debug('Seen a kwarg, assuming all following args are kw style')
```

Comment:
I asked this above, but is there a technical reason why we can't handle positional arguments after k/v pairs?

We're not even allowing spaces between the key portion, the equal sign, and the value portion, so we're not actually dealing with 3 arguments in the CLI, just the one. Why not allowing other arguments to come after?

Reply:
The reasoning is that once we have seen a kwarg and it is followed by a positional, we don't know what that positional argument is (since it is identified by its position in the arg list). This is consistent with python behaviour.

Reply:
I see the point, but aren't these being pre-handled by argparse so that they arrive as a dictionary as is? IIRC, and I may be conflating argparse and command parsing in the mgr module, arguments are identified by their dictionary k/v pairs rather than list positional arguments; no?

Reply:
Once we're in mgr module land, yes, we have the dict. But the dict is filled by position on the command line iiuc. But I need to look closer into ceph_argparse, the kwargs parsing should probably happen there.